draft-ietf-quic-tls.md

```diff
@@ -255,142 +205,206 @@ A simplified TLS 1.3 handshake with 0-RTT application data is shown in
+different encryption levels need to be sent, endpoints SHOULD use
+compound packets to send them in the same UDP datagram.
 
 
 ## Handshake and Setup Sequence
 
 The integration of QUIC with a TLS handshake is shown in more detail in
-{{quic-tls-handshake}}.  QUIC `STREAM` frames on stream 0 carry the TLS
-handshake.  QUIC performs loss recovery {{QUIC-RECOVERY}} for this stream and
-ensures that TLS handshake messages are delivered in the correct order.
+{{quic-tls-handshake}}.
 
 ~~~
-    Client                                             Server
+Client                                                   Server
 
-@H QUIC STREAM Frame(s) <0>:
-     ClientHello
-       + QUIC Extension
-                            -------->
-                        0-RTT Key => @0
-
-@0 QUIC STREAM Frame(s) <any stream>:
-   Replayable QUIC Frames
-                            -------->
-
-                                      QUIC STREAM Frame <0>: @H
-                                               ServerHello
-                                  {TLS Handshake Messages}
-                            <--------
-                        1-RTT Key => @1
-
-                                           QUIC Frames <any> @1
-                            <--------
-@H QUIC STREAM Frame(s) <0>:
-     (EndOfEarlyData)
-     {Finished}
-                            -------->
-
-@1 QUIC Frames <any>        <------->      QUIC Frames <any> @1
-~~~
-{: #quic-tls-handshake title="QUIC over TLS Handshake"}
+<CRYPTO_HS[
+        ClientHello]>  --------->
 
-In {{quic-tls-handshake}}, symbols mean:
+(STREAM[0-RTTData])    --------->
 
-* "<" and ">" enclose stream numbers.
+                       <---------   <ACK,
+                                        CRYPTO_HS[ServerHello]>
 
-* "@" indicates the keys that are used for protecting the QUIC packet (H =
-  handshake, using keys from the well-known cleartext packet secret;
-  0 = 0-RTT keys; 1 = 1-RTT keys).
+                       <---------  {CRYPTO_HS[
+                                           EncryptedExtensions,
+                                                   Certificate,
+                                             CertificateVerify,
+                                                     Finished]}
 
-* "(" and ")" enclose messages that are protected with TLS 0-RTT handshake or
-  application keys.
+                      <---------         {STREAM[0.5-RTT Data])
+{ACK,
+ CRYPTO_HS[Finished]} --------->
 
-* "{" and "}" enclose messages that are protected by the TLS Handshake keys.
+[Any frames]          <-------->                   [Any frames]
 
-If 0-RTT is not attempted, then the client does not send packets protected by
-the 0-RTT key (@0).  In that case, the only key transition on the client is from
-handshake packets (@H) to 1-RTT protection (@1), which happens after it sends
-its final set of TLS handshake messages.
+~~~
+{: #quic-tls-handshake title="QUIC Handshake"}
```

Comment:
Is it possible to include packet types in the above diagram, not only the key's symbols?

Reply:
I will add a new diagram.